File: backup_before_industrialisation_20250626/backend/metier_generator/templates/threed/tests_centralises/security/test_security.js
// Tests avancés pour la sécurité threed
const { checkAccess, auditLog, validateToken } = require('../../api/security');
describe('Sécurité threed', () => {
  it('doit refuser l’accès sans token', () => {
    expect(() => checkAccess(null)).toThrow();
  });
  it('doit valider un token JWT conforme', () => {
    const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...';
    expect(validateToken(token)).toBe(true);
  });
  it('doit enregistrer un audit log', () => {
    const result = auditLog({ action: 'login', user: 'admin' });
    expect(result).toHaveProperty('timestamp');
  });
});

File: backup_before_industrialisation_20250626/backend/metier_generator/templates/threed/tests_centralises/security/test_security.py
"""
Test centralisé security threed (Python).
"""

import pytest
from ...security import check_access, audit_log, validate_token


def test_access_denied_without_token():
    with pytest.raises(Exception):
        check_access(None)


def test_valid_jwt_token():
    token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9..."
    assert validate_token(token) is True


def test_audit_log():
    result = audit_log({"action": "login", "user": "admin"})
    assert "timestamp" in result
